src/content/content-events.js

```diff
@@ -0,0 +1,27 @@
+import { dispatchEvent, HEARTBEAT_EVENT, RESPONSE_PACKAGE_INFO_EVENT } from '../events-shared';
+
+let isWebappReady = false;
+export const waitForWebappReady = () => {
+  return new Promise((resolve) => {
+    const interval = setInterval(() => {
+      if (isWebappReady) {
+        clearInterval(interval);
+        resolve(true);
+      }
+    }, 100);
+
+    setTimeout(() => {
+      clearInterval(interval);
+      resolve(false);
+    }, 5000);
+  });
+};
```

Comment:
IMO no need to `resolve(false)` if you can call `reject()` (built-in in promise)
suggesting the following implementation with `setTimeout`

```suggestion
let isWebappReady = false;

function onAppLoaded(timeout = 5000, sleep = 100) {
  let started = Date.now();
  return new Promise((resolve, reject) => {
    const interval = () => {
      if (isWebappReady) {
        resolve();
        return;
      }
      let duration = Date.now() - started;
      if(duration >= timeout){
        reject(`could not load app. timeout exceeded after ${timeout} milliseconds`)
        return;
      }
      setTimeout(interval, sleep);
    };
    interval();
  });
}
```

example: https://replit.com/@jossef/wait-for-resource#index.js

Reply:
I don't see a significant difference, these are two different methods of performing the same task.

Resolved anyway.